pkg/cri/server/image_pull.go

```diff
@@ -548,20 +548,28 @@ const (
 	// the target image and will be passed to snapshotters for preparing layers in
 	// parallel. Skipping some layers is allowed and only affects performance.
 	targetImageLayersLabel = "containerd.io/snapshot/cri.image-layers"
+	// Secret or password used to resolve or pull container image layers.
+	targetRegistrySecret = "containerd.io/snapshot/cri.registry-secret"
```

Comment:
Annotations shouldn't be used to store/pass sensitive values such as secrets.

Reply:
> Annotations shouldn't be used to store/pass sensitive values such as secrets.

Yes. I also felt a little odd or hacky by passing secrets to the remote snapshotter from the Containerd this way. But I haven't figured out an elegant method yet. Do you have some suggestions about passing secrets to the remote snapshotter?

Reply:
I'd like to share my PoC ever for testing in Kubernetes:
- only set the secret namespace/name in the annotation
- runtime/snapshotter then retrieves the sensitive password/username or anything else from the secret by calling an additional binary, which only gets the content from the Kubernetes secret resources.

Reply:
> I'd like to share my PoC ever for testing in Kubernetes:
> 
> * only set the secret namespace/name in the annotation
> * runtime/snapshotter then retrieves the sensitive password/username or anything else from the secret by calling an additional binary, which only gets the content from the Kubernetes secret resources.
> 
> This is to say that add a hook mechanism on the snapshotter side.
Thanks for shareing @liubin 
I am curious how the hook binary can get Kubenetes serect? Do you deploy a kubeconfig on node?